includes/event-date.php

```
@@ -0,0 +1,173 @@
+<?php
+
+namespace Wporg\TranslationEvents;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeZone;
+use Exception;
+
+/**
+ * Event_Date
+ *
+ * The event date is in local time, get the UTC time via the utc() method.
+ *
+ * @package Wporg\TranslationEvents
+ */
+abstract class Event_Date extends DateTimeImmutable {
+	protected $event_timezone;
+	public function __construct( string $date, DateTimeZone $timezone = null ) {
+		if ( ! $timezone ) {
+			$timezone = new DateTimeZone( 'UTC' );
+		}
+
+		try {
+			$utc_date = new DateTime( $date, new DateTimeZone( 'UTC' ) );
+			$utc_date->setTimezone( $timezone );
+		} catch ( Exception $e ) {
+			$utc_date = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
```

I think maybe we should let it throw, otherwise this could result in an event being created/updated with a date that was not what the user supplied.

---

Also follow up material, I think.